build/upstream-builds/build-flink.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+#
+# This script supports:
+#
+#   Build to specific PNDA version
+#       Pass "PNDA" as first parameter
+#       Pass platform-salt branch or tag as second parameter (e.g. release/3.2)
+#   Build specific upstream version
+#       Pass "UPSTREAM" as first parameter
+#       Pass upstream branch or tag as second parameter (e.g. 1.2.3.4)
+
+MODE=${1}
+ARG=${2}
+
+EXCLUDES="-x test"
+set -e
+set -x
+
+export LC_ALL=en_US.utf8
+
+function error {
+    echo "Not Found"
+    echo "Please run the build dependency installer script"
+    exit -1
+}
+
+echo -n "Java 1.8: "
+if [[ $($JAVA_HOME/bin/javac -version 2>&1) != "javac 1.8"* ]]; then
+    error
+else
+    echo "OK"
+fi
+
+echo -n "shyaml: "
+if [[ -z $(which shyaml) ]]; then
+    error
+else
+    echo "OK"
+fi
+
+
+if [[ ${MODE} == "PNDA" ]]; then
+    FLINK_VERSION=$(wget -qO- https://raw.githubusercontent.com/pndaproject/platform-salt/${ARG}/pillar/services.sls | shyaml get-value flink.release_version)
+
+elif [[ ${MODE} == "UPSTREAM" ]]; then
+    FLINK_VERSION=${ARG}
+fi
+
+# Using apache-flink as a upstream project
+wget https://github.com/apache/flink/archive/release-${FLINK_VERSION}.tar.gz
+tar xzf release-${FLINK_VERSION}.tar.gz
+
+# Build upstream gobblin
+mkdir -p pnda-build
+
+
+# Build the package using maven
+cd flink-release-${FLINK_VERSION}
+for HADOOP_DISTRIBUTION in CDH HDP
+do
+    if [[ "${HADOOP_DISTRIBUTION}" == "CDH" ]]; then
+        HADOOP_VERSION=2.6.0-cdh5.9.0
+    fi
+    if [[ "${HADOOP_DISTRIBUTION}" == "HDP" ]]; then
+        HADOOP_VERSION=2.7.3.2.6.4.0-91
+    fi
+
+    mvn clean install -DskipTests -Pvendor-repos -Dhadoop.version="${HADOOP_VERSION}"
```

Comment:
Building Flink required Maven 3.0.4 (or higher) as defined in the documentation, would be good to check the version currently installed to ensure the build node is well setup (like checking java 1.8)
refer to https://github.com/pndaproject/pnda/blob/develop/build/install-build-tools.sh#L170

Reply:
Currently maven 3.0.5 is installed to the mirror which meets the dependency.
Reference - install-build-tools.sh ( line 168 )